src/base/InstanceId.h

```diff
@@ -58,14 +58,36 @@ class InstanceId
     } \
     template<> void \
     InstanceId<Class>::change() { \
-        static InstanceId<Class>::Value Last = 0; \
+        static auto Last = InstanceId<Class>::Value(); \
+        value = ++Last ? Last : ++Last; \
+    }
+
+/// 2-parameter instantiation macro (for private use only).
+/// In other sources use InstanceIdDefinitions() instead.
+#define InstanceIdDefinitions3(Class, ValueType, pfx) \
+    template<> const char * \
+    InstanceId<Class, ValueType>::prefix() const { \
+        return pfx; \
+    } \
+    template<> std::ostream & \
+    InstanceId<Class, ValueType>::print(std::ostream &os) const { \
+        return os << pfx << value; \
+    } \
+    template<> void \
+    InstanceId<Class, ValueType>::change() { \
+        static auto Last = InstanceId<Class, ValueType>::Value(); \
```

Comment:
Does C++ require that class specification prefix or can we do this instead:
```suggestion
        static auto Last = Value(); \
```

Reply:
Done